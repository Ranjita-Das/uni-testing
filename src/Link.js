import React from 'react'
import ReactDOM from 'react-dom'

class Link extends React.Component{
  constructor(props){
    super(props);
    this.state={
      class: "Come To Me",
    }
    this._onMouseEnter=this._onMouseEnter.bind(this);
    this._onMouseLeave=this._onMouseLeave.bind(this);
  }

  _onMouseEnter(){
    this.setState({
      class: "You Are On Me",
    })
  }

  _onMouseLeave(){
    this.setState({
      class: "Come To Me",
    })
  }

  render(){
    return(
      <a
        href={this.props.page || "#"}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        {this.state.class || "Link"}
      </a>
    )
  }
}

export default Link;
