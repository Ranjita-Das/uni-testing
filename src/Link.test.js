import React from 'react';
import Link from './Link.js';
import renderer from 'react-test-renderer';

it('the link renders correctly',()=>{
  const tree= renderer.create(<Link page="https://www.ranjitadas.com/" />).toJSON();
  expect(tree).toMatchSnapshot();
});
