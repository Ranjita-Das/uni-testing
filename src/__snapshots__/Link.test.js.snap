// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the link renders correctly 1`] = `
<a
  href="https://www.ranjitadas.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Come To Me
</a>
`;
